synapse/tests/test_lib_types.py

```diff
@@ -1,14 +1,67 @@
 # -*- coding: UTF-8 -*-
 import base64
 
-import synapse.cores.common as s_cores_common
 
 import synapse.common as s_common
 import synapse.lib.types as s_types
+import synapse.lib.xact as s_xact
 
 from synapse.tests.common import *
 
-class DataTypesTest(SynTest):
+
+class TestTypes(SynTest):
+
+    def test_hex_type(self):
+        with self.getTestCore() as core:
+            # fixme - getTestCore should have this dude loaded in him already!
+            modu = core.addCoreMods([('synapse.tests.test_cortex.TestModule', {})])
+
+            t = core.model.type('testhexa')
+            # Test norming & index
+            testvectors = [
+                ('C', 'c', b'\x0c'),
+                ('10', '10', b'\x10'),
+                ('10001', '10001', b'\x01\x00\x01'),
+                ('0x10001', '10001', b'\x01\x00\x01'),
+                ('FfF', 'fff', b'\x0f\xff'),
+                ('12A3e', '12a3e', b'\x01\x2a\x3e'),
+                (65537, '10001', b'\x01\x00\x01'),
+                (0, '0', b'\x00'),
+                (b'\x01\x00\x01', '10001', b'\x01\x00\x01'),
+                (b'\xd4\x1d\x8c\xd9\x8f\x00\xb2\x04\xe9\x80\t\x98\xec\xf8B~',
+                 'd41d8cd98f00b204e9800998ecf8427e',
+                 b'\xd4\x1d\x8c\xd9\x8f\x00\xb2\x04\xe9\x80\t\x98\xec\xf8B~'),
+                (-10, s_exc.BadTypeValu, None),
+                ('newp', s_exc.BadTypeValu, None)
+            ]
+
+            for v, n, b in testvectors:
+                if isinstance(n, str):
+                    r, subs = t.norm(v)
+                    self.eq(r, n)
+                    self.eq(subs, {})
+                    self.eq(t.indx(r), b)
+                else:
+                    self.raises(n, t.norm, v)
+
+            t = core.model.type('testhex4')
+            r, subs = t.norm('12A4')
+            self.eq(r, '12a4')
+            self.raises(s_exc.BadTypeValu, t.norm, '1234A')
```

Comment:
Would also show what happens when you norm something that isn't that wide

Reply:
we already have that - the `testhexa` form will norm any width

Reply:
I'm writing more extensive test vectors ot account for undersized ints where width > 1

Reply:
ok